### ISSUE TITLE
Primitive datatypes encoding and decoding tests

### ISSUE THREAD

**Issue by JoelBender** [JoelBender/bacpypes#38]
Add tests for primitive data encoding and decoding.

**Comment by JoelBender** [comment]
For the first crack at it with this commit https://github.com/JoelBender/bacpypes/commit/f6dadbf22ebc541ab8abcb743919a656b2f37149, I just added simple Integer object tests.

**Comment by JoelBender** [comment]
This commit https://github.com/JoelBender/bacpypes/commit/22c5fe48f303bf1431758f259b6ae6a68da0d6ff is in a series that adds lots of new tests.  They work correctly on py27 but not on py34 yet.

**Comment by JoelBender** [comment]
Found the problems with py34, NaN is tricky, and was masked in the py27 version by the way __cmp__ was implemented, which isn't used in py34.  https://github.com/JoelBender/bacpypes/commit/e12523eeeeb09d231798091fc659b73569cfbd6f

**Comment by JoelBender** [comment]
I think I deleted the branch called **issue-38** by mistake.  I wanted to keep it going because not all of the tests are done yet, and I still had the branch on my development machine, so I brought it up-to-date with *stage* and pushed it https://github.com/JoelBender/bacpypes/commit/a99e0f4664ca40a757b3cca8b1a7e6b1ded49b2e.

**Comment by JoelBender** [comment]
I merged this branch into `stage` because it's at a point where I can stop and switch to fixing the bugs that I've found.

**Comment by JoelBender** [comment]
As of 41a66ad92d63e9d7ecc3b32cebb842d343dcd250 the new date parsing code has been merged in.  The next step will be to make sure the tests are up to speed, merge it into stage, and release!

**Comment by JoelBender** [comment]
I added some more error checking and fiddled with the patterns a bit.  As for testing, I added a sample application that generates permutations of test strings.  With so many acceptable patterns and values, I've just been reviewing the output of the application and picking out a few at random to see if they make sense.  Maybe if I run the application with some code coverage tools I can reduce the permutations to a minimal subset that still provides good tests.
